### ISSUE TITLE
[EKS] [request]: add "cluster name" to metadata endpoint

### ISSUE THREAD

**Issue by jicowan** [aws/containers-roadmap#686]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Tell us about your request**
What do you want us to build?
A metadata endpoint for EKS clusters that allow applications running within the cluster to get information about a cluster without querying an AWS API.  The downward API only provides information about the pod. 

**Which service(s) is this request for?**
EKS

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
What outcome are you trying to achieve, ultimately, and why is it hard/impossible to do right now? What is the impact of not having this problem solved? The more details you can provide, the better we'll be able to understand and solve the problem.
I am trying to get the name of the cluster that a pod in running on from within the pod itself. 

**Are you currently working around this issue?**
How are you currently solving this problem?
I am still debating how to tackle this. 

**Additional context**
Anything else we should know?

**Attachments**
If you think you might have additional information that you'd like to include via an attachment, please do - we'll take a look. (Remember to remove any personally-identifiable information.)

**Comment by tabern** [comment]
@jicowan do you imagine more data than just the cluster name being exposed?

**Comment by jicowan** [comment]
Yes, @tabern.  It might be useful to include other attributes of the cluster, e.g. region, subnets, control plane security group id, etc.

**Comment by aviorma** [comment]
why it is not getting attention here ? this may be useful for us to operate deployment on EKS gathering this kind of details through EC2 METADATA, is there a different way today to gather this information when i am running on the EKS node ?

**Comment by tetianakravchenko** [comment]
Hi @tabern, any plans/updates on this issue?

**Comment by venom02** [comment]
any update? this would help for multi-cluster monitoring

**Comment by ottramst** [comment]
Hi!

Any updates on whether this is getting priority in the near future?

**Comment by Elasticjustin** [comment]
+1

**Comment by TonyLovesDevOps** [comment]
Hey y'all, this request may or may not be implemented, but you may be able to achieve the desired outcome by [querying IMDS to get the tags associated with the EC2 instance](https://aws.amazon.com/about-aws/whats-new/2022/01/instance-tags-amazon-ec2-instance-metadata-service/) in order to derive the cluster name.

For example:
```bash
$ curl http://169.254.169.254/latest/meta-data/tags/instance/aws:eks:cluster-name
foo
```

Just be sure to:
* Enable exposing instance metadata tags on the instances (disabled by default)
* Watch out for [this issue](https://github.com/terraform-aws-modules/terraform-aws-eks/issues/1785) if you use `/` characters in your tag keys. I changed mine to use `::` instead of `/` for delimiters.

**Comment by mikestef9** [comment]
Moving to coming soon

**Comment by collerek** [comment]
Is there a chance to include cluster id too?

**Comment by yosrixp** [comment]
Can you provide a native way for pods to access cluster metadata (e.g., cluster name, region, VPC ID) similar to the GKE Metadata Server? In GKE you can access the cluster's metadata from inside the POD while in EKS you can't access any of the metatdata.

**Comment by danielloader** [comment]
As an aside when I provision an EKS cluster with terraform these days I throw a cluster-metadata configmap in kube-system after provisioning with all these data points - it's not perfect obviously as you need to then replicate it to the right namespaces but it does in a pinch if you need to inject some values. 

Ideally it'd be accessible as both configmap (or secret) and API call, so you could still inject it as env vars where necessary.

**Comment by allir** [comment]
@mikestef9
> Moving to coming soon

Is soon coming soon?
